### DISCUSSION TITLE
Clarifying the Confusing Smoothing Settings for Sensor Noise

### DISCUSSION THREAD

**Discussion by utzaki** [NightscoutFoundation/xDrip#3709] [General]
Hi,

I’ve found that there are three settings related to smoothing sensor noise:

1. Other settings > Less common settings > Other misc. options > Smooth Libre data
2. Display settings > Smooth sensor noise
3. Display settings > Graph settings > Graph smoothing

I’m sure each of these has a different purpose, but it’s not clear to me, and I find it a bit confusing. Does anyone know the difference between them? Perhaps better labeling of the settings could help clarify their meanings.

**Comment by Navid200** [comment]
Yes, this clarification has been requested in facebook as well.  I hope we can do that by editing the user interface.

Graph smoothing affects the graph only.  It does not affect the data used to trigger alerts or data broadcasted to other apps.  So, I think the title is good in that respect.  I have gone into detail about that option only here:
https://navid200.github.io/xDrip/docs/Display/GraphSmoothing.html  

I cannot provide any info about the other two you have asked about.

I also know that there is a filter that was added or updated about a year ago that related to Libre only.  So, it could be one you are asking about.

**Comment by Navid200** [comment]
OK, I found the one that was added recently for Libre.  it is actually Libre 2:
https://github.com/NightscoutFoundation/xDrip/pull/2849

**Reply by utzaki** [reply]
Thanks @Navid200 , I saw your great documentation already before and this was very helpful to understand the **third option** (numbering is referring to my list in my post).

I could not find out, if the **second option** also affects alarms. That would be more than helpful, because high sensor noise sometimes is causing a false alarm. Especially at night time that is really annoying. 🥴 In case it is indeed smoothing values before they go into alarm routines, this setting should be moved somewhere outside of the "Display settings" section from my point of view.

The **first one** seems to be only relevant for Libreusers and is not relevant for me. But nevertheless also here clarifying may be useful. 

If you need ideas on how to clarify in the UX I could help giving some ideas. But before, I need a basic understanding of how every setting is affecting things.
